flask/app/tasks.py

```diff
@@ -1,84 +1,66 @@
 import os
 from datetime import datetime, timedelta
+from sqlalchemy.orm import joinedload
 from . import models
 from .email import send_email
-from .extensions import db
-from dataclasses import asdict
 
 
 def send_email_notification(app):
     with app.app_context():
-
-        app.logger.info("Sending email notification...")
-
         yesterday = (datetime.now() - timedelta(1)).strftime("%Y-%m-%d")
-        today = datetime.now().strftime("%Y-%m-%d")
 
-        analyses = db.session.query(models.Analysis).filter(
-            models.Analysis.requested >= yesterday,
-            models.Analysis.requested < today,
+        analyses = (
+            models.Analysis.query.options(joinedload(models.Analysis.datasets))
```

Comment:
It looks like later, the dictionary construction is incurring all sorts of additional queries. We could add all the joinedloads for each relationship here.